src/InstrumentationEngine.Lib/LoggerService.cpp

```diff
@@ -67,14 +67,12 @@ HRESULT CLoggerService::GetLoggingFlags(_Out_ LoggingFlags* pLoggingFlags)
 
     IfNotInitRetUnexpected(m_initialize);
 
-    CCriticalSectionHolder holder(&m_cs);
```

Comment:
While this is a good first step, don't you also need to move the lock acquisition in functions like LogMessage and others in order to take advantage of them?

Reply:
I think we still need the lock acquisition in those functions because they guard the logging sinks. This change ensures we  don't lock in LogMessage and quickly return if "Errors" are enabled.

Reply:
Hmm, that's not what https://github.com/microsoft/CLRInstrumentationEngine/blob/0ae92f3bd8249aa5eff8615a6f3d2b1602c5fbf1/src/InstrumentationEngine.Lib/LoggerService.cpp#L170-L180 seems to be doing

Reply:
CLogging checks GetLoggingFlags before calling LoggerService::LogMessage, but you're right, a call from some other place to LoggerService would hit this critical section.

Reply:
We might be able to speed things up some more by updating AllowLogEntry so that it can be called before the critical section. However, most callers don't go through this directly. They first get the flags from the logging service, check to see if the flag matches what they expect, and then call log. AllowLogEntry does a double-check. We have verification from Jakub that, under normal usage, this fixes his issue.

---
In reply to: [421841204](https://github.com/microsoft/CLRInstrumentationEngine/pull/258#discussion_r421841204) [](ancestors = 421841204)

Reply:
I'll add a task for addressing AllowLogEntry